File: problem/Reverse_Vowels/reverseVowels.js
/**
 * @param {string} s
 * @return {string}
 */
function reverseVowels(s) {
    const chars = s.split('');
    const n = chars.length;
    let low = 0;
    let high = n - 1;
    const vowels = new Set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']);
    
    while (low < high) {
        if (vowels.has(chars[low]) && vowels.has(chars[high])) {
            [chars[low], chars[high]] = [chars[high], chars[low]];
            low++;
            high--;
        } else {
            if (!vowels.has(chars[high])) {
                high--;
            }
            if (!vowels.has(chars[low])) {
                low++;
            }
        }
    }
    
    return chars.join('');
}

File: problem/Reverse_Vowels/reverseVowels.c
bool isVowel(char c)
{
    return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' ||
           c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U';
}

char *reverseVowels(char *s)
{
    int n = strlen(s);
    int low = 0;
    int high = n - 1;

    while (low < high)
    {
        if (isVowel(s[low]) && isVowel(s[high]))
        {
            // Swap vowels
            char temp = s[low];
            s[low] = s[high];
            s[high] = temp;
            low++;
            high--;
        }
        else
        {
            if (!isVowel(s[high]))
            {
                high--;
            }
            if (!isVowel(s[low]))
            {
                low++;
            }
        }
    }

    return s;
}
